### ISSUE TITLE
Compiler-generated code is analyzed using dotnet CLI RC3

### ISSUE THREAD

**Issue by martincostello** [DotNetAnalyzers/StyleCopAnalyzers#2282]
Using dotnet ```1.0.0-rc3-004530```, it appears that compiler-generated code is being analyzed and creating build warnings.

For example, a test project causes a ```Program.cs``` to be generated in the ```obj``` folder, which is then found and analyzed:

```
obj\Debug\netcoreapp1.1\API.Tests.Program.cs(2,15): warning SA1502: Element must not be on a single line [C:\api\tests\API.Tests\API.Tests.csproj]
```

Is there an MSBuild setting that can be added to exclude these folders from analysis while not affecting compilation?

**Comment by sharwell** [comment]
Can you show the first lines of **API.Tests.Program.cs** here? I haven't seen one of those files yet.

**Comment by martincostello** [comment]
Sure. It's essentially a 3 line file containing the below:

```csharp
using System;
class Program {static void Main(string[] args){}}

```

**Comment by martincostello** [comment]
I guess maybe the issue is more that it's that the RC3 SDK is not adding a header showing it's generated code?

**Comment by sharwell** [comment]
That sounds correct. Note that starting with the 1.1 release, this project no longer deals with exclusions *at all*. Roslyn is wholly responsible for identifying generated code and avoiding analysis in those files. From the information provided here, I would not be surprised if the 1.1 release produces the same result on that file.

**Comment by martincostello** [comment]
@sharwell Looks like it's an issue in vstest.

**Comment by martincostello** [comment]
Hopefully Microsoft/vstest#424 will fix it for version ```1.0.0``` of the SDK.

**Comment by martincostello** [comment]
Resolved by Microsoft/vstest#424.
